docs/docs/reference/frontend-styleguide.md

```diff
@@ -0,0 +1,423 @@
+---
+
+# Frontend Styleguide
+
+This part of the documentation contains reference material for the frontend styleguide, including design principles, UI components, and best practices.
+It is intended to be followed in all kitconcept projects and add-ons.
+
+## Use TypeScript
+
+We use TypeScript for all new frontend code.
+Typings does not have to be perfect and crystal correct, but this alone, should be better than using plain JavaScript.
+This helps to catch errors early and improves code quality.
+
+## Add-on configuration
+
+### Unified configuration structure
+
+We use a unified configuration folder structure for all kitconcept projects and add-ons.
+Update it for existing projects if necessary.
+We use `src/index.ts` for our add-ons (including project add-ons):
+
+```ts
+import type { ConfigType } from '@plone/registry';
+
+const applyConfig = (config: ConfigType) => {
+  // Apply the configuration
+
+  return config;
+};
+
+export default applyConfig;
+```
+
+We will separate the configuration into multiple files given their purpose.
+Create a `src/config` folder and add the necessary configuration files there.
+For example, we can have the following files:
+
+-   `src/config/settings.ts` for add-on specific configuration
+-   `src/config/blocks.ts` for blocks specific configuration
+-   `src/config/slots.ts` for slots specific configuration
+-   `src/config/widgets.ts` for widgets specific configuration
+
+Then, from the `src/index.ts` file, we can import and apply the configuration from these files:
+
+```ts
+import type { ConfigType } from '@plone/registry';
+import installSettings from './config/settings';
+import installBlocks from './config/blocks';
+import installSlots from './config/slots';
+import installWidgets from './config/widgets';
+
+const applyConfig = (config: ConfigType) => {
+  installSettings(config);
+  installBlocks(config);
+  installSlots(config);
+  installWidgets(config);
+
+  return config;
+};
+
+export default applyConfig;
+```
+
+and for each configuration file, we can have something like this:
+
+```ts
+import type { ConfigType } from '@plone/registry';
+
+const installBlocks = (config: ConfigType) => {
+  // Install the blocks configuration
+
+  return config;
+};
+
+export default installBlocks;
+```
+
+### Do not use spread operator for configuration
+
+It is not needed, we are not in Redux world in here.
+
+```ts
+// ❌ Avoid this
+config.blocks.blocksConfig = {
+  ...config.blocks.blocksConfig,
+  myBlock: {
+    // block configuration
+  },
+};
+```
+Instead, directly assign the configuration:
+
+```ts
+// ✅ Do this instead
+config.blocks.blocksConfig.myBlock = {
+  // block configuration
+};
+```
+
+Same for any other configuration, e.g. `config.widgets`, `config.settings`, etc.
+
+## No CSS imports as a side effect
+
+```tsx
+import './MyComponent.css'; // ❌ Avoid this
+```
+
+Instead, use VLT theme integration by adding `src/theme/_main.scss` and `src/theme/_variables.scss` files to your add-on.
```

Comment:
Link to the docs: https://6.docs.plone.org/volto/theming/theming-a-base-theme.html